dlp/snippets/inspect/inspect_hotword_rule.go

```diff
@@ -0,0 +1,130 @@
+
+package inspect
+
+// [START dlp_inspect_hotword_rule]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// inspectWithHotWordRules inspects data with hot word rule, it uses custom
+// regex with a hot word rule to increase the likelihood match
+func inspectWithHotWordRules(w io.Writer, projectID, textToInspect, customRegexPattern, hotWordRegexPattern string) error {
+	// projectID := "my-project-id"
+	// textToInspect := "Patient's MRN 444-5-22222 and just a number 333-2-33333"
+	// customRegexPattern := "[1-9]{3}-[1-9]{1}-[1-9]{5}"
+	// hotWordRegexPattern := "(?i)(mrn|medical)(?-i)"
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Construct the custom regex detectors
+	var customInfoType = &dlppb.CustomInfoType{
+		InfoType: &dlppb.InfoType{
+			Name: "C_MRN",
+		},
+		Type: &dlppb.CustomInfoType_Regex_{
+			Regex: &dlppb.CustomInfoType_Regex{
+				Pattern: customRegexPattern,
+			},
+		},
+		Likelihood: dlppb.Likelihood_POSSIBLE,
+	}
+
+	// Construct hotword rule.
+	var inspectionRuleSet = &dlppb.InspectionRuleSet{
+		Rules: []*dlppb.InspectionRule{
+			{
+				Type: &dlppb.InspectionRule_HotwordRule{
+					HotwordRule: &dlppb.CustomInfoType_DetectionRule_HotwordRule{
+						HotwordRegex: &dlppb.CustomInfoType_Regex{
+							Pattern: hotWordRegexPattern,
+						},
+						Proximity: &dlppb.CustomInfoType_DetectionRule_Proximity{
+							WindowBefore: int32(10),
+						},
+						LikelihoodAdjustment: &dlppb.CustomInfoType_DetectionRule_LikelihoodAdjustment{
+							Adjustment: &dlppb.CustomInfoType_DetectionRule_LikelihoodAdjustment_FixedLikelihood{
+								FixedLikelihood: dlppb.Likelihood_VERY_LIKELY,
+							},
+						},
+					},
+				},
+			},
+		},
+		InfoTypes: []*dlppb.InfoType{
+			customInfoType.InfoType,
+		},
+	}
+
+	// Construct the Inspect request to be sent by the client.
+	req := &dlppb.InspectContentRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/global", projectID),
+		Item:   contentItem,
+		InspectConfig: &dlppb.InspectConfig{
+			CustomInfoTypes: []*dlppb.CustomInfoType{
+				customInfoType,
+			},
+			RuleSet: []*dlppb.InspectionRuleSet{
+				inspectionRuleSet,
+			},
+			IncludeQuote: true,
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```

Comment:
issue: only return the error. The caller can inspect the error to understand what happened.

Reply:
fixed it.